app/src/main/java/com/alexafit/tasky/MainActivity.kt

```diff
@@ -3,49 +3,97 @@ package com.alexafit.tasky
 import android.os.Bundle
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
+import androidx.activity.viewModels
+import androidx.compose.foundation.layout.Box
 import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.foundation.layout.padding
 import androidx.compose.material.Scaffold
 import androidx.compose.material.rememberScaffoldState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.core.splashscreen.SplashScreen.Companion.installSplashScreen
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
 import androidx.navigation.compose.NavHost
 import androidx.navigation.compose.composable
 import androidx.navigation.compose.rememberNavController
 import com.alexafit.agendapresentation.overview.Overview
+import com.alexafit.coreui.components.loading.CircularLoadingDialog
 import com.alexafit.onboardingAuthPresentation.event.navigation.NavigationEvent
 import com.alexafit.onboardingAuthPresentation.login.LoginScreen
 import com.alexafit.onboardingAuthPresentation.register.RegisterScreen
 import com.alexafit.tasky.navigation.Route
 import com.alexafit.tasky.ui.theme.TaskyTheme
 import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.launch
 
 @AndroidEntryPoint
 class MainActivity : ComponentActivity() {
 
+    private val mainViewModel: MainViewModel by viewModels()
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        // need to check to see if user has active session
-        val activeSession = false
+        val splashScreen = installSplashScreen()
+        var uiState: MainActivityUiState by mutableStateOf(MainActivityUiState.Loading)
+
+        lifecycleScope.launch {
+            lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                mainViewModel.uiState.collect {
+                    uiState = it
+                }
+            }
+        }
+        splashScreen.setKeepOnScreenCondition {
+            when (uiState) {
+                MainActivityUiState.Loading -> true
+                MainActivityUiState.Success -> false
+            }
+        }
         setContent {
             TaskyTheme {
+                val modifier = Modifier
```

Comment:
What's the use of having one modifier and reuse it?

Each component should be independant of each other, so it's usually better to make each have it's own modiifer

Ex: Line 60, use: Modifier.fillMaxSize(), instead of modifier.fillMaxSize()